Tensile/Source/lib/include/Tensile/TensorDescriptor.hpp

```diff
@@ -258,26 +258,35 @@ namespace Tensile
     template <typename T>
     void WriteTensor(std::ostream & stream, T * data, TensorDescriptor const& desc)
     {
-        if(desc.dimensions() != 3)
-            throw std::runtime_error("Fix this function to work with dimensions != 3");
-
-        std::vector<size_t> index3{0,0,0};
-
-        stream << "Tensor("
-            << desc.sizes()[0] << ", "
-            << desc.sizes()[1] << ", "
-            << desc.sizes()[2] << ")";
-
-       stream << std::endl;
-
-        for(index3[2] = 0; index3[2] < desc.sizes()[2]; index3[2]++)
+        const size_t maxDims=8;
```

Comment:
Shouldn't be too difficult to extend this to arbitrary dimensions - see Reference.cpp in the client code, and CoordCount / CoordNumbered.

Reply:
I added a comment to indicate we can make this better.  The PR is an improvement over previous support which was hard-corded to only 3 dims.